tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -1757,11 +1758,8 @@ def friedrich_coefficients(x, param):
         assert coeff >= 0, "Coefficients must be positive or zero. Found {}".format(coeff)
 
         # calculate the current friedrich coefficients if they do not exist yet
-        if m not in calculated:
-            calculated[m] = {r: _estimate_friedrich_coefficients(x, m, r)}
-        else:
-            if r not in calculated[m]:
-                calculated[m] = {r: _estimate_friedrich_coefficients(x, m, r)}
+        if m not in calculated or r not in calculated[m]:
+            calculated[m].update({r: _estimate_friedrich_coefficients(x, m, r)})
```

Comment:
Instead of dict of dicts, an alternative would be to use a single dict with tuples `(m, r)` as keys, let me know if you prefer this solution.

Reply:
I am fine with the dict of dicts, but maybe you can write 
`calculated[m][r] = ...`
as I think it is more clear. Maybe you have a different opinion.

Reply:
Agreed, much more readable, thanks.